library/std/src/sys/pal/uefi/helpers.rs

```diff
@@ -142,7 +148,50 @@ pub(crate) unsafe fn close_event(evt: NonNull<crate::ffi::c_void>) -> io::Result
 
 /// Get the Protocol for current system handle.
 /// Note: Some protocols need to be manually freed. It is the callers responsibility to do so.
-pub(crate) fn image_handle_protocol<T>(protocol_guid: Guid) -> Option<NonNull<T>> {
-    let system_handle = uefi::env::try_image_handle()?;
-    open_protocol(system_handle, protocol_guid).ok()
+pub(crate) fn image_handle_protocol<T>(protocol_guid: Guid) -> io::Result<NonNull<T>> {
+    let system_handle = uefi::env::try_image_handle().ok_or(io::const_io_error!(
+        io::ErrorKind::NotFound,
+        "Protocol not found in Image handle"
+    ))?;
+    open_protocol(system_handle, protocol_guid)
+}
+
+pub(crate) fn device_path_to_text(path: *mut device_path::Protocol) -> io::Result<OsString> {
+    let device_path_to_text_handles = locate_handles(device_path_to_text::PROTOCOL_GUID)?;
+    for handle in device_path_to_text_handles {
```

Comment:
nit: I don't think a loop is needed here, as there should only be one device-path-to-text handle and protocol instance.

Reply:
Well, logically, yes, there should only be one of this protocol. However, the spec doesn't make any such assertion so is there any reason to not to use the loop?

Reply:
Even if an implementation did have multiple handles for this protocol, I don't see how you'd get to a situation where opening the first one fails, but opening the second one succeeds. Not a big deal either way though -- I don't think the loop hurts anything, just seems unnecessary to me.

Reply:
Well, I would prefer to leave the loop if that is alright. I have added a cache for the handle similar to https://github.com/rust-lang/rust/pull/120889 since, as you said, the handle for this protocol will probably be constant.

Reply:
Sure, I'm OK with keeping the loop.